wallet/wallet-controller/src/sync/mod.rs

```diff
@@ -92,6 +98,19 @@ pub async fn sync_once<T: NodeInterface>(
             wallet.best_block().map_err(ControllerError::WalletError)?;
 
         if chain_info.best_block_id == wallet_block_id {
+            // if everything is synced up scan the mempool
+            let transactions: Vec<_> = rpc_client
+                .get_all_mempool_transactions()
```

Comment:
This isn't really a good idea. The mempool can potentially have hundreds of thousands of transactions. At best this will take forever, and at worst this will timeout your RPC layer and will make your whole sync function fail.

This needs some smart design where we use some kind of pub/sub to get this information. Now the problem is that our RPC doesn't inherently support pub/sub since it's http based (which Pavel got around in the wallet for blocks by polling), so the only solution (I have in mind) is to use some kind of (bloom, or otherwise) filter to tell the mempool that we're interested in some specific transactions, and then keep polling for them. This is more complex than it looks and has more complex design requirements. Pavel has implemented a bloom filter for p2p already, btw. But bloom isn't the only way to do this.

Another solution is to make the wallet launch a p2p node. This way it can receive transactions like nodes and filter them based on whether it's interested in their outcome. This is what light wallets do, like Electrum.

This is one of the reasons why I was asking to postpone this whole problem. It's unclear how we're gonna solve it and it's absolutely not a priority. The only thing that gives this a priority is making the wallet nicer, where users feel that the node is snappy (in receiving transactions), despite it being deceiving since transactions will show up in the wallet while not confirmed yet.

What I recommend now is putting this PR aside, and working on whatever is prior on the queue. Once we decide to work on this problem, we should come up with a reasonable design that takes everything into consideration.

Reply:
removed the sync from mempool